var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -196,16 +196,12 @@ def check_install(self):
             with open("check.c", 'w') as f:
                 f.write(source)
             if '+mpi' in spec:
-                cc = which('%s' % spec['mpi'].mpicc)
+                cc = which(spec['mpi'].mpicc)
```

Comment:
I would use `Executable` for these instead of `which` since you already know the path.